lib/ClangImporter/ImportType.cpp

```diff
@@ -421,6 +421,22 @@ namespace {
       return Type();
     }
 
+    ImportResult VisitDynamicRangePointerType(
+        const clang::DynamicRangePointerType *type) {
+      // DynamicRangePointerType is a clang type representing a pointer with
+      // an "ended_by" type attribute for -fbounds-safety. For now, we don't
+      // import these into Swift.
+      return Type();
```

Comment:
Shouldn't we recurse on the wrapped type here, since it's okay to ignore the `ended_by` attribute? We'll be gathering this information separately.

Reply:
Right. Though this is to make them consistent with how `CountAttributedType` is handled in this codebase and downstream. @hnrklssn's WIP PR will fix the visitor for `CountAttributedType ` later.

Reply:
If you prefer, I can make `DynamicRangePointerType` and `ValueTerminatedType` to be handled like other `SUGAR_TYPE` here. Please let me know.

Reply:
I would prefer that they be handled like other `SUGAR_TYPE`. The bounds-safety annotations shouldn't break interop, even when they aren't being used to improve it.

Reply:
Fixed!

Reply:
@hnrklssn You will have to remove `SUGAR_TYPE(CountAttributed)` in your patch and reintroduce `Visit CountAttributed`. I fixed it anyway knowing that your patch won't be landing in the release branch, but this will be.

Reply:
@DougGregor actually, I couldn't land `SUGAR_TYPE(DynamicRangePointer)` because `DynamicRangePointerType` doesn't exist in Clang yet. I will revert back to `return Type()` and then write up a follow up patch to fix that right after I land https://github.com/swiftlang/llvm-project/pull/9665. 

```
swift/lib/ClangImporter/ImportType.cpp:943:5: error: member access into incomplete type 'const clang::DynamicRangePointerType'
  943 |     SUGAR_TYPE(DynamicRangePointer)
      |     ^
swift/lib/ClangImporter/ImportType.cpp:930:24: note: expanded from macro 'SUGAR_TYPE'
  930 |       return Visit(type->desugar());                                           \
      |                        ^
```